### ISSUE TITLE
Possible breakage with gzipped csv files in 0.8 and 0.8.1

### ISSUE THREAD

**Issue by igorcalabria** [duckdb/duckdb#8204]
### What happens?

On 0.8 and 0.81

```sql
D SELECT * FROM read_csv('file.gz', auto_detect=TRUE);
Error: IO Error: Input is not a GZIP stream
```

On 0.7.1 it works fine

```sql
D SELECT * FROM read_csv('file.gz', auto_detect=TRUE);
<data_here> 
```

### To Reproduce

Unfortunately, I can't provide the file that triggers this issue, but I'm happy to provide more debug information. I just need some pointers(specific flags to compile, where to place breakpoints, which variables to dump, etc).

File extracts and tests (`gzip -t`) fine with gzip.

### OS:

linux

### DuckDB Version:

0.8 and 0.8.1

### DuckDB Client:

cli

### Full Name:

See profile

### Affiliation:

See profile

### Have you tried this on the latest `master` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

**Comment by igorcalabria** [comment]
@Tishj 
I had some time to come back this. The exception is thrown when verifying the gzip header, here: https://github.com/duckdb/duckdb/blob/master/src/common/gzip_file_system.cpp#L315 

The odd thing is that this is happening after almost the whole file has  been read. The code is assuming that there's another gzip file concatenated after the first stream which is not true in this case. I didn't have much time to dig deeper into the code but my guess is that there's something wrong with the buffer end checks around here: https://github.com/duckdb/duckdb/blob/master/src/common/gzip_file_system.cpp#L203 

This was probably introduced by: https://github.com/duckdb/duckdb/pull/6817, commit: 2574c7e558ddcc2db72f7d795d0c9cbe6c9c3de9

Please let me know if you need more information.

**Comment by Tishj** [comment]
I don't think we can investigate this without a reproduction example, if you could share the problematic file or create a reproduction file that also triggers this issue we'd be happy to debug this.

**Comment by igorcalabria** [comment]
@Tishj I've created a PR that potentially address this problem(at least all existing tests pass and it fixes the issue). 

This issue is triggered when the compressed stream ends with a empty buffer. In the debugger I can see that

- `sd.in_buff_end - (sd.in_buff.get() + sd.in_buf_size)` is equal to 0
- `mz_stream_ptr->avail_in` is equal to 4

with those conditions, neither of checks in

```c++
		// Last read from file done and remaining bytes only for footer or less
		if ((sd.in_buff_end < sd.in_buff.get() + sd.in_buf_size) && mz_stream_ptr->avail_in <= GZIP_FOOTER_SIZE) {
			Close();
			return true;
		}
		if (mz_stream_ptr->avail_in > GZIP_FOOTER_SIZE) {
			// Definitely not concatenated gzip
			if (*(sd.in_buff_start + GZIP_FOOTER_SIZE) != 0x1F) {
				Close();
				return true;
			}
		}
		// Concatenated GZIP potentially coming up - refresh input buffer
		sd.refresh = true;
```

are going to `Close()` and return that the file is finished. 

I think at least one more Read from the child stream is need to really know if the file ended or not, that's why I just defer the footer size check to the next iteration. 

To generate a test case for this, I think you need to either craft a  compressed stream that ends in the conditions described or play around with the buffer sizes.

**Comment by igorcalabria** [comment]
I've manged to reproduce this.  Set BUFFER_SIZE to 41 in https://github.com/duckdb/duckdb/blob/master/src/include/duckdb/common/gzip_file_system.hpp#L17 and run `SELECT count(*) FROM read_csv_auto('data/csv/tenk.tsv.gz');`

To find this, I made the buffer value configurable  
```patch
diff --git a/src/common/gzip_file_system.cpp b/src/common/gzip_file_system.cpp
index a1c351166d..238d51bc41 100644
--- a/src/common/gzip_file_system.cpp
+++ b/src/common/gzip_file_system.cpp
@@ -392,7 +392,11 @@ unique_ptr<StreamWrapper> GZipFileSystem::CreateStream() {
 }
 
 idx_t GZipFileSystem::InBufferSize() {
-	return BUFFER_SIZE;
+	if (const char* buffer_from_env = std::getenv("DUCKDB_CSV_IN_BUFFER")) {
+		return std::stol(buffer_from_env);
+	} else {
+		return BUFFER_SIZE;
+	}
 }
 
 idx_t GZipFileSystem::OutBufferSize() {

```

and ran this script 

```bash
#!/usr/bin/env bash

set -eo pipefail

for i in $(seq 32 1024)
do
    echo "SIZE: $i"
    export DUCKDB_CSV_IN_BUFFER="$i"
    ./build/duckdb -c "SELECT COUNT(*) FROM read_csv_auto('data/csv/tenk.tsv.gz')"
done
```
